components/services/tor/public/interfaces/BUILD.gn

```diff
@@ -7,4 +7,20 @@ mojom("interfaces") {
   public_deps = [
     "//mojo/public/mojom/base",
   ]
+  cpp_typemaps = [
+    {
+      types = [
+        {
+          mojom = "tor.mojom.TorConfig"
+          cpp = "tor::TorConfig"
+        },
+      ]
+      traits_headers = [ "//brave/common/tor/tor_config_struct_traits.h" ]
+      traits_sources = [ "//brave/common/tor/tor_config_struct_traits.cc" ]
+      traits_public_deps = [
+        "//mojo/public/cpp/base",
```

Comment:
we also need `//brave/common/tor` in  `traits_public_deps`

Reply:
see upstream changes of `proxy_mojom_traits` in https://chromium.googlesource.com/chromium/src/+/4b3aa3c059810d00033e43de06d5fe856da7ab3d%5E%21/

Reply:
@darkdh, if I do that I get
<pre>
ERROR Dependency cycle:
  //brave/components/services/tor/public/interfaces:interfaces ->
  //brave/common/tor:tor ->
  //brave/components/services/tor/public/interfaces:interfaces
</pre>

Reply:
we should remove `//brave/components/services/tor/public/interfaces:interfaces` in "//brave/common/tor". That dependency seems unnecessary

Reply:
Sounds good. Fixed in 657403db7ae13f4f1d02df3cf34390ae33584266